ElectricityBillApi/Models/ElectricProvider.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+
+namespace ElectricityBillApi.Models
+{
+    public class ElectricProvider
+    {
+        private PowerPlant _subscibedPowerPlant;
+
+        public string Name { get; set; }
+
+        public decimal CalculatePrice(Address address)
+        {
+            if (_subscibedPowerPlant == default) throw new NotImplementedException("What to do here? More infos needed!");
+
+            double priceDistanceFactor = CalculateDistanceFactor(address);
+            return _subscibedPowerPlant.ElectricityPrice * (decimal)priceDistanceFactor;
+        }
+
+        public void Subscribe(PowerPlant powerPlant)
+        {
+            if (_subscibedPowerPlant is null)
+            {
+                _subscibedPowerPlant = powerPlant;
+                return;
+            }
+
+            throw new InvalidOperationException("Povider already subscribed to a power plant.");
```

Comment:
It's nice to know which provider is already subbed. So include the name of a provider in an exception.

Reply:
Ok, done.., My initial thought was to not send out potential sensitive information